aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -858,6 +858,33 @@ TORCH_IMPL_FUNC(index_add_cpu_out)
     if (numel == 0) {
       return;
     }
+    if (dim == (result.dim() - 1) && alpha.equal(1.0) &&
+        index_contig.scalar_type() == ScalarType::Long &&
+        // Heuristic value that scatter_add has higher performance.
+        result.numel() >= (4 * at::internal::GRAIN_SIZE / at::get_num_threads())) {
```

Comment:
Could you please explain the logic of this heuristics?

Reply:
Use `result.numel() > at::internal::GRAIN_SIZE` instread to make sure `scatter_add`  has enough data to get relatively good performance.